File: dotnet/src/EPostak/Resources/EnterpriseResource.cs
namespace EPostak.Resources;

/// <summary>Workflow-first namespace for Enterprise <c>/api/v1/*</c> resources.</summary>
public sealed class EnterpriseResource
{
    public AuthResource Auth { get; }
    public AuditResource Audit { get; }
    public DocumentsResource Documents { get; }
    public InboxResource Inbox { get; }
    public FirmsResource Firms { get; }
    public PeppolResource Peppol { get; }
    public WebhooksResource Webhooks { get; }
    public ReportingResource Reporting { get; }
    public ExtractResource Extract { get; }
    public AccountResource Account { get; }
    public IntegratorResource Integrator { get; }
    public ConnectorResource Connector { get; }
    public EnterprisePullResource Pull { get; }

    internal EnterpriseResource(EPostakClient client)
    {
        Auth = client.Auth;
        Audit = client.Audit;
        Documents = client.Documents;
        Inbox = client.Documents.Inbox;
        Firms = client.Firms;
        Peppol = client.Peppol;
        Webhooks = client.Webhooks;
        Reporting = client.Reporting;
        Extract = client.Extract;
        Account = client.Account;
        Integrator = client.Integrator;
        Connector = client.Connector;
        Pull = new EnterprisePullResource(client.Inbound, client.Outbound);
    }
}

public sealed class EnterprisePullResource
{
    public InboundResource Inbound { get; }
    public OutboundResource Outbound { get; }

    internal EnterprisePullResource(InboundResource inbound, OutboundResource outbound)
    {
        Inbound = inbound;
        Outbound = outbound;
    }
}

File: java/src/main/java/sk/epostak/sdk/resources/EnterpriseResource.java
package sk.epostak.sdk.resources;

import sk.epostak.sdk.EPostak;

/** Workflow-first namespace for Enterprise /api/v1 resources. */
public final class EnterpriseResource {
    private final EPostak client;
    private final EnterprisePullResource pull;

    public EnterpriseResource(EPostak client) {
        this.client = client;
        this.pull = new EnterprisePullResource(client.inbound(), client.outbound());
    }

    public DocumentsResource documents() { return client.documents(); }
    public InboxResource inbox() { return client.documents().inbox(); }
    public FirmsResource firms() { return client.firms(); }
    public PeppolResource peppol() { return client.peppol(); }
    public WebhooksResource webhooks() { return client.webhooks(); }
    public ReportingResource reporting() { return client.reporting(); }
    public ExtractResource extract() { return client.extract(); }
    public AccountResource account() { return client.account(); }
    public AuthResource auth() { return client.auth(); }
    public BoxResource box() { return client.box(); }
    public ConnectorResource connector() { return client.connector(); }
    public AuditResource audit() { return client.audit(); }
    public IntegratorResource integrator() { return client.integrator(); }
    public EnterprisePullResource pull() { return pull; }
}
